### PR TITLE
Fix `torch.[size|stride]`(dim=None)` invocation

### LINKED ISSUE #111944: `torch.tensor.size(dim=None)` fails, expects `dim` to be a "name"
### 🐛 Describe the bug

`tensor.size()` doesn't behave the same as `tensor.size(dim=None)`, though docs that seem to indicate it should. The default value of `dim` is described as `None` here:

https://pytorch.org/docs/stable/generated/torch.Tensor.size.html

When passing None explicitly, it seems to be interpreting `dim` as if it's a name for a named tensor. cc: @zou3519 

```
In [8]: t = torch.randn( (3,4,5) )

In [9]: t.size()
Out[9]: torch.Size([3, 4, 5])

In [10]: t.size(dim=0)
Out[10]: 3

In [11]: t.size(dim=None)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In[11], line 1
----> 1 t.size(dim=None)

RuntimeError: Please look up dimensions by name, got: name = None.

In [12]: t.size(None)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In[12], line 1
----> 1 t.size(None)

RuntimeError: Please look up dimensions by name, got: name = None.

In [14]: t.size("foo")
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In[14], line 1
----> 1 t.size("foo")

RuntimeError: Name 'foo' not found in Tensor[None, None, None].
```

### Versions

PyTorch version: 2.2.0.dev20230922
Is debug build: False
CUDA used to build PyTorch: 12.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.6 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: version 3.16.3
Libc version: glibc-2.31

Python version: 3.11.5 (main, Sep 11 2023, 13:54:46) [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.15.0-1037-aws-x86_64-with-glibc2.31
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   46 bits physical, 48 bits virtual
CPU(s):                          32
On-line CPU(s) list:             0-31
Thread(s) per core:              2
Core(s) per socket:              16
Socket(s):                       1
NUMA node(s):                    1
Vendor ID:                       GenuineIntel
CPU family:                      6
Model:                           85
Model name:                      Intel(R) Xeon(R) Platinum 8175M CPU @ 2.50GHz
Stepping:                        4
CPU MHz:                         2500.006
BogoMIPS:                        5000.01
Hypervisor vendor:               KVM
Virtualization type:             full
L1d cache:                       512 KiB
L1i cache:                       512 KiB
L2 cache:                        16 MiB
L3 cache:                        33 MiB
NUMA node0 CPU(s):               0-31
Vulnerability Itlb multihit:     KVM: Mitigation: VMX unsupported
Vulnerability L1tf:              Mitigation; PTE Inversion
Vulnerability Mds:               Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown
Vulnerability Meltdown:          Mitigation; PTI
Vulnerability Mmio stale data:   Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown
Vulnerability Retbleed:          Vulnerable
Vulnerability Spec store bypass: Vulnerable
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Retpolines, STIBP disabled, RSB filling, PBRSB-eIBRS Not affected
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single pti fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid mpx avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves ida arat pku ospke
Versions of relevant libraries:
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.24.3
[pip3] torch==2.2.0.dev20230922
[pip3] torchaudio==2.2.0.dev20230922
[pip3] torchvision==0.17.0.dev20230922
[pip3] triton==2.1.0
[conda] blas                      1.0                         mkl
[conda] brotlipy                  0.7.0           py311h9bf148f_1002    pytorch-nightly
[conda] cffi                      1.15.1          py311h9bf148f_3    pytorch-nightly
[conda] cryptography              38.0.4          py311h46ebde7_0    pytorch-nightly
[conda] filelock                  3.9.0                   py311_0    pytorch-nightly
[conda] libjpeg-turbo             2.0.0                h9bf148f_0    pytorch-nightly
[conda] mkl                       2021.4.0           h06a4308_640
[conda] mkl-service               2.4.0           py311h9bf148f_0    pytorch-nightly
[conda] mkl_fft                   1.3.1           py311hc796f24_0    pytorch-nightly
[conda] mkl_random                1.2.2           py311hbba84a0_0    pytorch-nightly
[conda] mpmath                    1.2.1                   py311_0    pytorch-nightly
[conda] numpy                     1.24.3          py311hc206e33_0
[conda] numpy-base                1.24.3          py311hfd5febd_0
[conda] packaging                 22.0                    py311_0    pytorch-nightly
[conda] pillow                    9.3.0           py311h3fd9d12_2    pytorch-nightly
[conda] pluggy                    1.0.0                   py311_1    pytorch-nightly
[conda] pysocks                   1.7.1                   py311_0    pytorch-nightly
[conda] pytorch                   2.2.0.dev20230922 py3.11_cuda12.1_cudnn8.9.2_0    pytorch-nightly
[conda] pytorch-cuda              12.1                 ha16c6d3_5    pytorch-nightly
[conda] pytorch-mutex             1.0                        cuda    pytorch-nightly
[conda] requests                  2.28.1                  py311_0    pytorch-nightly
[conda] tomli                     2.0.1                   py311_0    pytorch-nightly
[conda] torchaudio                2.2.0.dev20230922     py311_cu121    pytorch-nightly
[conda] torchtriton               2.1.0+6e4932cda8           py311    pytorch-nightly
[conda] torchvision               0.17.0.dev20230922     py311_cu121    pytorch-nightly
[conda] urllib3                   1.26.14                 py311_0    pytorch-nightly

cc @zou3519 @albanD

### PR DESCRIPTION
**Pull request by malfet**
Per documentation, one should be able to explicitly pass dim argument as None to get tensor size across all dimentions/strides, but before this change it was incorrectly interpreted as named tensor call.

Modify `size` and `stride` signatures generated by `gen_pyi.py` to highlight that overload with `None` will return a Tuple, but one with `dim: _int` returns `int`.

Add regression test to validate the behavior, and remove the check for asserts from two named tensors tests (NamedTensors are dead, aren't they?)

Fixes https://github.com/pytorch/pytorch/issues/111944

### CODE REVIEW

File: test/test_misc.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Owner(s): ["module: tests"]
```
Comment (zou3519) [inline test/test_misc.py:1]: I would put this into test/test_torch.py, that is where a lot of our basic tensor tests go

**Review by zou3519 — APPROVED** [review]

File: test/test_misc.py (hunk 1 above)
Comment (albanD) [reply test/test_misc.py:1]: I would agree that a test_misc is not very descriptive. Also no team is signed up to look at module: tests so I think we should avoid using it here as much as possible.

**Review by albanD — COMMENTED** [review]
nit on the test file name. but SGTM otherwise

File: test/test_misc.py (hunk 1 above)
Comment (malfet) [reply test/test_misc.py:1]: Drumroll (this is where I got the idea in the first place):
https://github.com/pytorch/pytorch/blob/ed15fa7cc2adf9918e716d669ec0ed77c488fc69/test/test_torch.py#L1


And this is why I've picked this one for new file, but sure, can move it to `test_torch.py`

**Pull request closed without merging**